tests/models/test_model_input_examples.py

```diff
@@ -163,6 +185,11 @@ def test_input_examples(pandas_df_with_all_types, dict_of_ndarrays):
         x = _Example(example)
         x.save(tmp.path())
         filename = x.info["artifact_path"]
+        assert x.info["type"] == "dataframe"
+        assert x.info["pandas_orient"] == "split"
+        with open(tmp.path(filename)) as f:
+            parsed_json = json.load(f)
+            pd.read_json(json.dumps(parsed_json), orient=x.info["pandas_orient"])
```

Comment:
For these read instances, can we just do a quick assert frame equals check to the original data that was used to generate the example to be 100% confident that we're reloading in a consistent manner with the assigned orientation type?

Reply:
That won't make sense in an example like this - here the example that was passed wasn't a DataFrame in the first place, so reading it back via pd.read_json will not give the original data back. This will hopefully be handled better by the work being planned to handle plain Python objects

Reply:
ah, right. For the externalized new test that validates the selective behavior, can we perform some degree of validation of orient type content unification (just to make sure that the dynamic conversion logic preserves content and inferred types)?

Reply:
Yes, working on it now!

Reply:
Generally done with this as well. The main sticking point that came up is that if a user has a `values` orient DF, when it gets converted to an example, everything gets stuck under a `data` key. That gets interpreted as a column header if the user tries to load it back directly with read_json with a values orient. Hence the note in the test about pulling out data via the `data` key.

Reply:
That's completely fine. That orientation isn't really something that MLflow supports anyway (casting won't work for signature enforcement through serving, some other things don't work correctly with that format either).